stac-example/src/main/scala/geotrellis/stac/raster/StacAssetReprojectRasterSource.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.stac.raster
+
+import geotrellis.proj4.{CRS, Transform}
+import geotrellis.raster.io.geotiff.OverviewStrategy
+import geotrellis.raster._
+import geotrellis.vector.Extent
+import geotrellis.raster.reproject.{Reproject, ReprojectRasterExtent}
+import geotrellis.raster.resample.ResampleMethod
+import cats.syntax.option._
+
+class StacAssetReprojectRasterSource(
+  val asset: StacAsset,
+  val crs: CRS,
+  val resampleTarget: ResampleTarget = DefaultTarget,
+  val resampleMethod: ResampleMethod = ResampleMethod.DEFAULT,
+  val strategy: OverviewStrategy = OverviewStrategy.DEFAULT,
+  val errorThreshold: Double = 0.125,
+  private[geotrellis] val targetCellType: Option[TargetCellType] = None,
+  @transient underlyingRS: => Option[RasterSource] = None
+) extends RasterSource {
+  @transient private lazy val underlying            = underlyingRS.getOrElse(RasterSource(asset.href))
+  @transient private lazy val underlyingReprojected = underlying.reproject(crs, resampleTarget, resampleMethod, strategy)
```

Comment:
This access forces `underlying` if and only if `underlyingReprojected` is forced right?

Reply:
@jisantuc affirmative! two underylings are neccesary to get access to the non modified metadata.